### PR TITLE
[16.04] Provide backward compatible galaxy.datatypes.checkers module...

### PR DESCRIPTION
**Pull request by jmchilton**
... for external datatypes.

xref http://dev.list.galaxyproject.org/STACKS-error-td4669428.html

### CODE REVIEW

File: lib/galaxy/datatypes/checkers.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"""Module proxies :module:`galaxy.util.checkers` for backward compatiblity.
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/checkers.py]: s/compatiblity/compatibility/

**Comment by jmchilton** [comment]
@nsoranzo You reviewed this earlier - any chance I can get you to +1 it and merge?

File: lib/galaxy/datatypes/checkers.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+)
+
+__all__ = [
+    'check_binary',
+    'check_bz2',
+    'check_gzip',
+    'check_html',
+    'check_image',
+    'check_zip',
+    'HTML_CHECK_LINES',
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/checkers.py]: Not sure `HTML_CHECK_LINES` should be exported.

**Comment by martenson** [comment]
why did Travis fail to run this?

**Comment by martenson** [comment]
runs here
https://travis-ci.org/galaxyproject/galaxy/builds/132283289

**Comment by dannon** [comment]
@jmchilton Did you want to address the HTML_CHECK_LINES point prior to a merge?

Otherwise, this looks fine to me as-is; exposing HTML_CHECK_LINES won't _break_ anything, though I hope no consumer tools are actually using that.

**Comment by jmchilton** [comment]
I'll revise this in order to not expose that - the impulse of limiting what is exposed is good.

**Comment by dannon** [comment]
@jmchilton I agree with the sentiment, but if this is only being added back for compatibility, it should expose exactly what was there before the move to util.compat, right?  (which was everything)

**Comment by jmchilton** [comment]
@dannon Well we are defining the interface we providing backward compatibility as we go here. Certainly I'd prefer if none of this stuff had been used and datatypes didn't even exist in the tool shed - at least not without a defined and maintainable interface to Galaxy.  But the pragmatic reality is they did exist - we shouldn't provide backward compatibility to more than we need to in order to fix actual exported datatypes though in my opinion.

**Comment by dannon** [comment]
@jmchilton Hrmm, was looking at this more like a 16.04 bugfix, but since it's just HTML_CHECK_LINES it's not worth making the case for -- seems like this crept a bit, to me, though.

**Comment by nsoranzo** [comment]
:+1: Thanks @jmchilton!

**Pull request merged**